fe/fe-core/src/main/java/com/starrocks/connector/elasticsearch/ElasticsearchMetadata.java

```diff
@@ -0,0 +1,107 @@
+import com.starrocks.connector.ConnectorMetadata;
+import com.starrocks.connector.elasticsearch.external.EsRestClient;
+import com.starrocks.connector.elasticsearch.external.EsUtil;
+import com.starrocks.sql.ast.DropTableStmt;
+import com.starrocks.sql.optimizer.OptimizerContext;
+import com.starrocks.sql.optimizer.operator.scalar.ColumnRefOperator;
+import com.starrocks.sql.optimizer.operator.scalar.ScalarOperator;
+import com.starrocks.sql.optimizer.statistics.Statistics;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+import static com.starrocks.connector.hive.HiveMetastoreApiConverter.CONNECTOR_ID_GENERATOR;
+
+// TODO add meta cache
+public class ElasticsearchMetadata
+        implements ConnectorMetadata {
+    private final EsRestClient esRestClient;
+    private final Map<String, String> properties;
+    private final String catalogName;
+    public static final String DEFAULT_DB = "default";
+    public static final long DEFAULT_DB_ID = 1L;
+
+    public ElasticsearchMetadata(EsRestClient esRestClient, String catalogName, Map<String, String> properties) {
+        this.esRestClient = esRestClient;
+        this.catalogName = catalogName;
+        this.properties = properties;
+    }
+
+    @Override
+    public List<String> listDbNames() {
+        return Arrays.asList(DEFAULT_DB);
+    }
+
+    @Override
+    public List<String> listTableNames(String dbName) {
+        return esRestClient.listTables();
+    }
+
+    @Override
+    public Database getDb(String dbName) {
+        return new Database(DEFAULT_DB_ID, DEFAULT_DB);
+    }
+
+    @Override
+    public Table getTable(String dbName, String tblName) {
+        if (!DEFAULT_DB.equalsIgnoreCase(dbName)) {
+            return null;
+        }
+        return toEsTable(esRestClient, properties, catalogName, tblName);
+    }
+
+    @Override
+    public Statistics getTableStatistics(OptimizerContext session,
+                                         Table table,
+                                         Map<ColumnRefOperator, Column> columns,
+                                         List<PartitionKey> partitionKeys,
+                                         ScalarOperator predicate) {
+        return null;
+    }
+
+    public static EsTable toEsTable(EsRestClient esRestClient,
+                                    Map<String, String> properties,
+                                    String catalogName,
+                                    String tableName) {
+        List<Column> columns = EsUtil.convertColumnSchema(esRestClient, tableName);
+        try {
+            properties.put(EsTable.INDEX, tableName);
+            EsTable esTable = new EsTable(CONNECTOR_ID_GENERATOR.getNextId().asInt(),
+                    tableName,
+                    columns,
+                    properties,
+                    new SinglePartitionInfo(),
+                    catalogName);
+            esTable.syncTableMetaData(esRestClient);
```

Comment:
How time-consuming is es sync. should we cache the es table?

Reply:
If there are a lot of indexes, we may get all tables and sync all tables for each time when executing use database statement.

Reply:
no,  it's Estable member function,  so it will only sync itself